### DISCUSSION TITLE
Force 'colum' layout for text in stat panel

### DISCUSSION THREAD

**Discussion by RutgerKe** [grafana/grafana#40269] [Dashboards, Panels & Visualizations]
I would like to show a couple of values in a [stat](https://grafana.com/docs/grafana/latest/visualizations/stat-panel/) panel. Depending on a variable selected, some values are not available. The prometheus queries are implemented with `> 0`, so they do not show up when they are not relevant, this works well. However, the layout could be better.

Ideally I would like the numbers displayed horizontally, with the label on top. Example when all (in this case 3) numbers are available, which looks good:
![image](https://user-images.githubusercontent.com/6214906/136767410-26e63ab5-48a3-42df-9f3d-c5cd051c628c.png)

When two of the values are hidden the label is moved right next to it, which would look better on top.
![image](https://user-images.githubusercontent.com/6214906/136767470-cc298e16-0493-477d-b1d6-445708e4456b.png)

Resizing the panel to be smaller (too small for three numbers, but just to illustrate what it should look like, but with  more blank space left and right):
![image](https://user-images.githubusercontent.com/6214906/136767595-3f52a32f-d43b-40dc-b262-813f896c1779.png)

At first I thought this had to do with the options 'horizontally' and 'vertically', but I think this just concerns the values relative to each other, and not the label.

**What I think could be a solution**: An option to select if you always want the label in 'column' layout with the number. Another option would be to link it to the existing 'horizontal'/ 'vertical' layout, but I'm not sure if there are any side effects to that.

The current workaround I use is to not use 'center text', which looks slightly better, but is still less than ideal:
![image](https://user-images.githubusercontent.com/6214906/136768004-226055d2-84b4-4e7e-a27b-9f0dfb3bcde3.png)

Let me know if you need any more information! Thanks

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
